master/internal/stream/system.go

```diff
@@ -103,10 +145,24 @@ type FilterMaker[T stream.Msg, S any] interface {
 	MakeFilter() func(T) bool
 }
 
-func (ps PublisherSet) Start(ctx context.Context) {
-	// start each publisher
-	go publishLoop(ctx, "stream_trial_chan", ps.Trials)
-	// go publishLoop(ctx, "stream_experiment_chan", ps.Experiments)
+func start[T stream.Msg](
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	channel string,
+	publisher *stream.Publisher[T],
+) {
+	publishLoop(ctx, channel, publisher)
+	wg.Done()
+}
+
+// Start starts each Publisher in the PublisherSet.
+func (ps *PublisherSet) Start(ctx context.Context) error {
+	var wg sync.WaitGroup
+	wg.Add(1) // XXX: hard-coding this will cause issues if not updated correctly
+	go start(ctx, &wg, trialChan, ps.Trials)
```

Comment:
use another errorsx.Group?

This is another instance of exactly what it's for.

Reply:
Awesome! I had some hesitation to startup up additional groups because of the child contexts; however, that really shouldn't be an issue since cancellations get propagated down to children